### PR TITLE
feat(TD-10867): Filter with DSEL expression using valid and invalid functions

### PR DESCRIPTION
**Pull request by amaumont**
**What is the problem this Pull Request is trying to solve?**

- converting a TQL expression using 'valid' or 'invalid'  to a DSEL expression is not supported
 
**What is the chosen solution to this problem?**

- convert a TQL expression using 'valid' or 'invalid' to a DSEL expression
- needs as input the native type or semantic type of each field to be able to provide it as parameter of the DSEL functions 'isValid()' or 'isInvalid()'
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/DSELConverter.java [hunk 1]
```diff
@@ -8,17 +8,20 @@
...
 
     /**
      * Utility method to convert a TQL query to a DSEL query.
      *
      * @param query TQL query as String
+     * @param fieldToType the mapping to get the type for each field
      * @return DSEL ELNode ready to serve for DSEL intepreter
      */
-    public static ELNode convert(String query) throws TqlException {
+    public ELNode convert(String query, Map<String, String> fieldToType) throws TqlException {
```
Comment (amaumont) [inline daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/DSELConverter.java:23]: `fieldToType`:
- do you have a better proposal about the name `fieldToType`
- for the moment this is the most basic implementation for this kind of data, not sure we will need something more advanced at this moment.

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.talend.tqldsel.DSELConverter.wrapNode;
 
 public class DSELVisitorFunctionCallTest {
 
+    static Map<String, String> fieldToType;
+
+    @BeforeAll
+    static void setUp() {
+        final HashMap<String, String> fToType = new HashMap<>();
```
Comment (amaumont) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java:22]: Should we gather this setup code in an util class shared with the other test classes `DSELVisitorNotTest`, `DSELVisitorOrAndTest`, and `DSELVisitorSimpleComparisonTest`, or gather the other classes content in `DSELVisitorFunctionCallTest`?

File: daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/DSELConverter.java [hunk 3]
```diff
@@ -8,28 +8,34 @@
...
     /**
      * Utility method to convert a TQL query to a DSEL query.
      *
      * @param tqlQuery TQL query
+     * @param fieldToType a Map object used to get a type (native or semantic type) from a field name, this is a lightweight
+     * representation of the schema
      * @return DSEL ELNode ready to serve for DSEL interpreter
      */
-    public static ELNode convert(final Expression tqlQuery) throws TqlException {
+    public static ELNode convert(final Expression tqlQuery, Map<String, String> fieldToType) throws TqlException {
```
Comment (abenali-talend) [inline daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/DSELConverter.java:37]: can you add a signature for expressions that do not need the map
```suggestion
    public static ELNode convert(final Expression tqlQuery) throws TqlException
     {
        return convert(tqlQuery, Map.of());
      }
     
```

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java (hunk 2 above)
Comment (amaumont) [reply daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java:22]: Will be done in an other PR.

**Comment by amaumont** [comment]
This pull request is replaced by https://github.com/Talend/daikon/pull/882 which starts from the branch https://github.com/Talend/daikon/tree/amu-fwa/feat/TDP-11256/dsel-to-tql

**Pull request closed without merging**